### PR TITLE
Remove tear-off mocking in refresh_test

### LINKED ISSUE #15849: `flutter test` fails on latest engine (https://github.com/flutter/flutter/pull/15847)
```
00:20 +96 ~1 -1: loading C:\projects\flutter sdk\packages\flutter\test\cupertino\scaffold_test.dart
00:21 +96 ~1 -1: C:\projects\flutter sdk\packages\flutter\test\cupertino\refresh_test.dart: UI tests - doesn't invoke anything without user interaction [E]
  NoSuchMethodError: The method '_setExpected' was called on null.
  Receiver: null
  Tried calling: _setExpected(Closure: (Invocation) => dynamic)
  dart:core                             Object.noSuchMethod
  package:mockito/src/mock.dart 318:15  PostExpectation._completeWhen
  package:mockito/src/mock.dart 304:12  PostExpectation.thenReturn
  cupertino\refresh_test.dart 27:30     main.<fn>
  
  'package:flutter_test/src/binding.dart': Failed assertion: line 519 pos 12: 'inTest': is not true.
  dart:core                                     _AssertionError._throwNew
  package:flutter_test/src/binding.dart         TestWidgetsFlutterBinding.postTest
  package:flutter_test/src/binding.dart 680:11  AutomatedTestWidgetsFlutterBinding.postTest
 ```

### PR DESCRIPTION
**Pull request by srawlins**
Fixes #15849.

In Dart 2 semantics (so far...) Mock objects cannot have their methods torn off as a part of mocking.

Related to https://github.com/dart-lang/build/issues/4842.

### CODE REVIEW

**Comment by srawlins** [comment]
CC @xster

**Comment by srawlins** [comment]
CC @aam

**Review by aam — APPROVED** [review]

**Comment by srawlins** [comment]
Learning things. This PR breaks compatibility with Dart 1 mode. Working on a fix that works in both modes.

**Comment by xster** [comment]
Still LG. Thanks for figuring it out!

**Comment by srawlins** [comment]
Green Travis. Go for it!

**Comment by srawlins** [comment]
Someone with write access should _totally_ merge this PR 😁 😁 😁

**Comment by aam** [comment]
Didn't realize you can't submit it, @srawlins .
AppVeyor was red so I restarted it, hopefully by the time it finished, the tree turns green too.

**Comment by aam** [comment]
AppVeyor completed, ready to be submitted as soon as flutter-build turns green.

**Comment by a-siva** [comment]
Can we submit this PR.

**Comment by aam** [comment]
Please go ahead

**Pull request merged**
